sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/FluxAutoLockRenew.java

```diff
@@ -168,7 +172,9 @@ protected void hookOnNext(ServiceBusMessageContext messageContext) {
             } catch (Exception e) {
                 logger.info("Exception occurred while handling downstream onNext operation.", e);
             } finally {
-                lockCleanup.accept(messageContext);
+                if (isAutoCompleteEnabled) {
+                    lockCleanup.accept(messageContext);
+                }
```

Comment:
Why do we need to check `isAutoCompleteEnabled` to clean up the auto-renew lock?

Reply:
isAutoCompleteEnabled  : If user has disabled autoComplete, we need lock renew to continue , we do not want to clean up in this case. When lock renew expire, it will be cleaned at that time.